feat(azure): Add support for Azure API Management

ekifel opened the pull request:
Issue #705

tim775 commented on internal/providers/terraform/azure/api_management.go:
```diff
@@ -0,0 +1,118 @@
+		capacity, _ = decimal.NewFromString(s[1])
+	}
+
+	if tier != "Consumption" {
```
I guess we need to do a case insensitive comparison

tim775 commented on internal/providers/terraform/azure/api_management.go:
```diff
@@ -0,0 +1,118 @@
+			Service:       strPtr("API Management"),
+			ProductFamily: strPtr("Developer Tools"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "skuName", Value: strPtr(tier)},
```
And also these filter look ups should be case insensitive (with `ValueRegex`) since tier is a user input.

tim775 reviewed (changes requested):
Looks good! two things:

- Change to case-insensitive for comparisons and price lookups (same as we discussed on CDN).
- Could you add a comment to PR indicating that you've double checked the cost calcs against the pricing page?  I'm open to ideas here so you can do something similar to what I did on the CDN pull request or something that makes more sense to you.

ekifel commented:
Checks the cost calcs from golden file against the [pricing page](https://azure.microsoft.com/en-us/pricing/details/api-management/):

azurerm_api_management.basic2 
- [x] API management (Basic); 2  units; $294.34

azurerm_api_management.consumption                                                                     
- [x] API management (consumption over 1M); 9  1M calls; $31.50  

azurerm_api_management.developer1                                                                      
- [x] API management (Developer); 1  units; $48.03 

azurerm_api_management.premium                                                                         
- [x] API management (Premium); 4  units; $11,180.68  
- [x] Self hosted gateway; 5  gateways; $5,000.50  

azurerm_api_management.premium_non_usage                                                               
- [x] API management (Premium); 4  units; $11,180.68  
- [x] Self hosted gateway; Monthly cost depends on usage: $1,000.10 per gateways    

azurerm_api_management.std3                                                                            
- [x] API management (Standard); 3  units; $2,060.13 

Total cost: 
- [x] $29,795.86

Link to Azure pricing calculator: https://azure.com/e/f4b3ade5e7e44107bc0b18a932eb2b63

ekifel commented:
Thanks @tim775! I updated comparison and regexes. Added comment with checks the cost calcs against the pricing page. 
Azure Pricing Calculator give users opportunity to share his estimate with another users, so, I attached a link to my calculations, you can take a quick look at this.

tim775 commented:
@alikhajeh1 LGTM

alikhajeh1 commented on internal/providers/terraform/azure/api_management.go:
```diff
@@ -0,0 +1,118 @@
+
+func consumptionAPICostComponent(location, tier string, quantity *decimal.Decimal) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            "API management (consumption over 1M)",
```
https://azure.microsoft.com/en-us/pricing/details/api-management/ mentions that "Monthly free grant of 1 million requests per Azure subscription".

We've been ignoring the per-account free tiers as the discounts are usually tiny, they slightly complicate the logic/output, and it's not possible for us to tell if this TF project will benefit from them or not. So maybe we can do the same here and remove the `(consumption over 1M)` and the logic above that subtracts 1M?

ekifel commented:
@alikhajeh1, I removed free tier, so it is ready for final review/merge
